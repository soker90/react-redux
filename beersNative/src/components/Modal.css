.modal {
  position: absolute;
  width: 100%;
  height: 100%;
  background: rgba(255,255,255,0.7);
  z-index: 100;
  top: 0;
  left: 0;
}

.modal-content {
  width: 600px;
  margin: 10% auto;
  border-radius: 3px;
  border: 1px solid #121212;
  background: #fff;
  padding: 5px;
}

.modal-header, .modal-body {
  border-bottom: 1px solid #121212;
}

.modal-footer {
  padding: 10px 0;
  height: 20px;
}

.modal-footer button {
  float: right;
}
