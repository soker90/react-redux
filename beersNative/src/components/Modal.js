import React, { Component } from 'react'
import './Modal.css'

class Modal extends Component {
  closeModal (){
    this.props.closeModal()
  }

  closeModalCallBack() {
    this.props.callback(this.props.extraParams)
    this.closeModal()
  }

  render() {
    const {title, body} = this.props

    return (
      <div className="modal">
        <div className="modal-content">
          <div className="modal-header">
            <h4>{title}</h4>
          </div>
          <div className="modal-body">
            <p>{body}</p>
          </div>
          <div className="modal-footer">
            {(this.props.callback !== null) && <button onClick={() => this.closeModalCallBack()}>Accept</button>}
            <button onClick={() => this.closeModal()}>Close</button>
          </div>
        </div>
      </div>
    )
  }

}

export default Modal
