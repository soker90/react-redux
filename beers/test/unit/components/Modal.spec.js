// Modal.spec.js
import React from 'react'
import renderer from 'react-test-renderer'

import Modal from '../../../src/components/Modal'

describe('Modal component', () => {
  test('render Modal component without crashing', () => {
    const title = 'Hey a modal title'
    const body = 'This is the modal body text'

    const tree = renderer.create(
      <Modal title={title} body={body} callback={null}/>
    )
    expect(tree.toJSON()).toMatchSnapshot()
  })

  test('render Modal component with callback', () => {
    const title = 'Hey a modal title'
    const body = 'This is the modal body text'
    const callback = jest.fn()

    const tree = renderer.create(
      <Modal title={title} body={body} callback={callback}/>
    )
    expect(tree.toJSON()).toMatchSnapshot()
  })

})
