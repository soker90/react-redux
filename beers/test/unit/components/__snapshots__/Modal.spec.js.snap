exports[`Modal component render Modal component with callback 1`] = `
<div
  className="modal">
  <div
    className="modal-content">
    <div
      className="modal-header">
      <h4>
        Hey a modal title
      </h4>
    </div>
    <div
      className="modal-body">
      <p>
        This is the modal body text
      </p>
    </div>
    <div
      className="modal-footer">
      <button
        onClick={[Function]}>
        Accept
      </button>
      <button
        onClick={[Function]}>
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`Modal component render Modal component without crashing 1`] = `
<div
  className="modal">
  <div
    className="modal-content">
    <div
      className="modal-header">
      <h4>
        Hey a modal title
      </h4>
    </div>
    <div
      className="modal-body">
      <p>
        This is the modal body text
      </p>
    </div>
    <div
      className="modal-footer">
      <button
        onClick={[Function]}>
        Close
      </button>
    </div>
  </div>
</div>
`;
